### PR TITLE
Replace DateTime with Timestamp

### PR DESCRIPTION
**Pull request by shinfan**
Fixes https://github.com/GoogleCloudPlatform/google-cloud-java/issues/1920

### CODE REVIEW

File: google-cloud-core/src/test/java/com/google/cloud/TimestampTest.java [hunk 1]
```diff
@@ -55,6 +57,13 @@ public void maxValue() {
   }
 
   @Test
+  public void fromMicroseconds() {
```
Comment (garrettjonesgoogle) [inline google-cloud-core/src/test/java/com/google/cloud/TimestampTest.java]: more accurate name: `ofMicroseconds`

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java [hunk 2]
```diff
@@ -730,46 +729,4 @@ public void testDelete() {
     assertNull(keys.next());
     assertFalse(keys.hasNext());
   }
-
-  @Test
-  public void testRunInTransaction() {
```
Comment (garrettjonesgoogle) [inline google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java]: Why is this test being deleted? (Neo just added it)

File: google-cloud-core/src/main/java/com/google/cloud/Timestamp.java [hunk 3]
```diff
@@ -74,6 +79,37 @@ public static Timestamp ofTimeSecondsAndNanos(long seconds, int nanos) {
   }
 
   /**
+   * Creates an instance representing the value of {@code microseconds}
+   *
+   * @throws IllegalArgumentException if the timestamp is outside the representable range
+   */
+  public static Timestamp ofTimeMicroseconds(long microseconds) {
+    long seconds = microseconds / 1000000L;
```
Comment (garrettjonesgoogle) [inline google-cloud-core/src/main/java/com/google/cloud/Timestamp.java]: Use `TimeUnit.MICROSECONDS.toSeconds(microseconds)`, and a similar type of conversion on the next line, and in the `of(Date)` function.

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java (hunk 2 above)
Comment (shinfan) [inline google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java]: oops it was a bad merge. Done

File: google-cloud-core/src/main/java/com/google/cloud/Timestamp.java (hunk 3 above)
Comment (shinfan) [inline google-cloud-core/src/main/java/com/google/cloud/Timestamp.java]: Done

**Comment by shinfan** [comment]
PTAL

File: google-cloud-core/src/main/java/com/google/cloud/Timestamp.java [hunk 4]
```diff
@@ -74,6 +79,38 @@ public static Timestamp ofTimeSecondsAndNanos(long seconds, int nanos) {
   }
 
   /**
+   * Creates an instance representing the value of {@code microseconds}
```
Comment (garrettjonesgoogle) [inline google-cloud-core/src/main/java/com/google/cloud/Timestamp.java]: add a period at the end of the line.

File: google-cloud-core/src/main/java/com/google/cloud/Timestamp.java [hunk 5]
```diff
@@ -74,6 +79,38 @@ public static Timestamp ofTimeSecondsAndNanos(long seconds, int nanos) {
...
+    long seconds = TimeUnit.MICROSECONDS.toSeconds(microseconds);
+    int nanos = (int) TimeUnit.MICROSECONDS.toNanos(
+        microseconds - TimeUnit.SECONDS.toMicros(seconds));
+    checkArgument(
+        Timestamps.isValid(seconds, nanos), "timestamp out of range: %s, %s", seconds, nanos);
+    return new Timestamp(seconds, nanos);
+  }
+
+  /**
+   * Creates an instance representing the value of {@code Date}
```
Comment (garrettjonesgoogle) [inline google-cloud-core/src/main/java/com/google/cloud/Timestamp.java]: period at the end

**Pull request merged**